tests/test_axis_limits.py

```diff
@@ -332,3 +336,57 @@ def test(eq):
 
         test(get("W7-X"))
         test(get("NCSX"))
+
+
+def _reverse_mode_unsafe_names():
+    names = data_index["desc.equilibrium.equilibrium.Equilibrium"].keys()
+    eq = get("ESTELL")
+
+    def isalias(name):
+        return isinstance(
+            data_index["desc.equilibrium.equilibrium.Equilibrium"][name]["fun"],
+            functools.partial,
+        )
+
+    def get_source(name):
+        return "".join(
+            inspect.getsource(
+                data_index["desc.equilibrium.equilibrium.Equilibrium"][name]["fun"]
+            ).split("def ")[1:]
+        )
+
+    names = [
+        name
+        for name in names
+        if not (
+            "Boozer" in name
+            or "_mn" in name
+            or name == "B modes"
+            or _skip_this(eq, name)
+            or name in not_finite_limits
+            or isalias(name)
+        )
+    ]
+
+    unsafe_names = []  # things that might have nan gradient but shouldn't
+    for name in names:
+        source = get_source(name)
+        if "replace_at_axis" in source:
+            unsafe_names.append(name)
+
+    unsafe_names = sorted(unsafe_names)
+    print("Unsafe names: ", unsafe_names)
+    return unsafe_names
+
+
+@pytest.mark.parametrize("name", _reverse_mode_unsafe_names())
+def test_reverse_mode_ad_axis(name):
+    """Asserts that the rho=0 axis limits are reverse mode differentiable."""
+    eq = get("ESTELL")
+    grid = LinearGrid(rho=0.0, M=2, N=2, NFP=eq.NFP, sym=eq.sym)
+    eq.change_resolution(2, 2, 2, 4, 4, 4)
+
+    obj = ObjectiveFunction(GenericObjective(name, eq, grid=grid), verbose=0)
+    obj.build(verbose=0)
```

Comment:
We only need to check if the reverse mode diff works for all quantities that are using safediv.  When you build a GenericObjective, you are not specifiying any specific quantity to compute. Is jax automatically building a graph across all the registered quantities?

Reply:
the `name` here is a parameter of the test, generated by the helper function above. We basically check every quantity that uses the `replace_at_axis` method, which usually means we're dividing by zero and then overwriting it.